### PR TITLE
[plain_hasher] Migrate to 2018 edition

### PR DESCRIPTION
**Pull request by koushiro**
# What I have changed

- Migrate `plain_hasher` crate to 2018 edition (related issue: #143)
- Replace `libtest` with `criterion`

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -34,7 +34,7 @@ script:
     fi
   - cd fixed-hash/ && cargo test --all-features && cd ..
   - cd uint/ && cargo test --features=std,quickcheck --release && cd ..
-  - cd plain_hasher/ && cargo test --no-default-features && cd ..
+  - cd plain_hasher/ && cargo test --no-default-features && cargo test --benches && cd ..
```
Comment (ordian) [inline .travis.yml]:
```suggestion
  - cd plain_hasher/ && cargo test --no-default-features && cargo check --benches && cd ..
```

Comment (koushiro) [reply .travis.yml]: It just my personal convention. `cargo test --benches` is building all benchmarkable targets and running them as tests (include running the unit-tests). Maybe we can do `cargo test --benches -- --bench` which will ignore running unit-tests.

**Comment by ordian** [comment]
@koushiro please merge with master

File: .travis.yml (hunk 1 above)
Comment (koushiro) [reply .travis.yml]: Actually, it runs the benchmarks like that (include running unit-tests):
```
running 1 test
test tests::it_works ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

...

Testing write_plain_hasher
Success

Testing write_default_hasher
Success
```

Comment (dvdplm) [reply .travis.yml]: Right, and we do not wish to run the benchmarks on CI.

(Not that it's a bad thing per se to run the benches, but it's not worth anything if we don't collect the output performance data and store it somewhere, and we're not so for the time being running the benches just slows CI down for no benefit.)

**Pull request merged**